driver/src/price_finding/linear_optimization_price_finder.rs

```diff
@@ -106,7 +106,7 @@ fn parse_token(key: &str) -> Result<u16, PriceFindingError> {
 fn parse_price_value(value: &serde_json::Value) -> Result<u128, PriceFindingError> {
     value
         .as_str()
-        .ok_or_else(|| PriceFindingError::from("Price value not a string"))?
+        .unwrap_or("0")
```

Comment:
I think this is not what we want here. I think we should handle the different JSON value types:
```rust
match value {
    Value::String(value) => { /* parse same as before */ }
    Value::Number(value) => { /* here value is a serde_json::Number which can be converted to u128 */ }
    Value::Null => 0,
    other => return PriceFindingError::new(format!("Expected numerical value but got '{}'", other), ErrorKind::JsonError),
}
```